ecoservice_financeinterface/models/ecofi.py

```diff
@@ -163,43 +163,46 @@ def set_main_account(self, cr, uid, move, context=None):
         a. Test if there is an invoice connected to the move_id and test if the invoice
             account_id is in the move than this is the mainaccount
         """
+        def first(set_element):
+            return next(iter(set_element))
+
         context = context or dict()
         ecofikonto = False
-        sollkonto = list()
-        habenkonto = list()
-        nullkonto = list()
+        sollkonto = set()
+        habenkonto = set()
+        nullkonto = set()
         error = False
         ecofikonto_no_invoice = move.line_id[0].account_id
 
         for line in move.line_id:
             Umsatz = Decimal(str(line.debit)) - Decimal(str(line.credit))
             if Umsatz < 0:
-                habenkonto.append(line.account_id)
+                habenkonto.add(line.account_id)
             elif Umsatz > 0:
-                sollkonto.append(line.account_id)
+                sollkonto.add(line.account_id)
             else:
-                nullkonto.append(line.account_id)
+                nullkonto.add(line.account_id)
         if len(sollkonto) == 1 and len(habenkonto) == 1:
             ecofikonto = move.line_id[0].account_id
         elif len(sollkonto) == 1 and len(habenkonto) > 1:
-            ecofikonto = sollkonto[0]
+            ecofikonto = first(sollkonto)
```

Comment:
Seems like you could just `sollkonto.pop()` it, but I guess the iteration method is more safe.